concent_api/gatekeeper/views.py

```diff
@@ -154,6 +158,30 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
     if len(transfer_token_paths_to_files) != len(set(transfer_token_paths_to_files)):
         return gatekeeper_access_denied_response('File paths in the token must be unique', path_to_file, loaded_golem_message.subtask_id, client_public_key)
 
+    if any((file["checksum"] is None or len(file["checksum"]) == 0 or file["checksum"].isspace()) for file in loaded_golem_message.files):
+        return gatekeeper_access_denied_response('Wrong type of checksum variable. The variable is empty', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+    if any(not isinstance(file["checksum"], str) for file in loaded_golem_message.files):
+        return gatekeeper_access_denied_response('Wrong type of checksum variable. The variable must be string', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+    file_checksums = [tuple(file["checksum"].split(":")) for file in loaded_golem_message.files]
+    if any(file_checksum[0] != hashing_algorytm.value for file_checksum in file_checksums for hashing_algorytm in HashingAlgorytm):
+        return gatekeeper_access_denied_response('Wrong type of checksum variable. The hashing algorytm is not supported', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+    if any(VALID_SHA1_HASH.match(file_checksum[1]) is None for file_checksum in file_checksums):
+        return gatekeeper_access_denied_response('Wrong type of checksum variable. The hash is incorrect', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+
+    if any((file["size"] is None) for file in loaded_golem_message.files):
+        return gatekeeper_access_denied_response('Wrong type of size variable. The variable is empty', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+    file_sizes = [file["size"] for file in loaded_golem_message.files]
+    for file_size in file_sizes:
+        try:
+            int(file_size)
+        except ValueError:
+            return gatekeeper_access_denied_response('Wrong type of size variable. The variable must be digit', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        except TypeError:
+            return gatekeeper_access_denied_response('Wrong type of size variable. The variable must be integer', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+
+    if any(int(file["size"]) < 0  for file in loaded_golem_message.files):
+        return gatekeeper_access_denied_response('Wrong type of size variable. The variable must contain negative digit', path_to_file, loaded_golem_message.subtask_id, client_public_key)
```

Comment:
The message is wrong. The type is OK. It's just that the size cannot be negative.